File: xauth-bean/src/main/java/org/xhome/xauth/AuthLog.java
package org.xhome.xauth;

import org.xhome.common.Base;

/**
 * @project xauth-bean
 * @author jhat
 * @email cpf624@126.com
 * @date Aug 9, 201311:12:32 PM
 * @description 用户认证信息
 */
public class AuthLog extends Base {

	private static final long	serialVersionUID	= 3418695337307713490L;
	private User 	user; 	 // 用户名
	private String 	method;  // 认证方式
	private String 	address; // 访问地址(IPv4/IPv6)
	private Short 	agent;   // 0:Other,1:Chrome, 2:Firefox, 3:IE, 4:Android
	private String 	number;  // 设备编号
	
	public AuthLog () {}
	
	public AuthLog (User user, String method, String address, Short agent, String number) {
		this.setUser(user);
		this.setMethod(method);
		this.setAddress(address);
		this.setAgent(agent);
		this.setNumber(number);
	}

	public User getUser() {
		return user;
	}

	public void setUser(User user) {
		this.user = user;
	}

	public String getMethod() {
		return method;
	}

	public void setMethod(String method) {
		this.method = method;
	}

	public String getAddress() {
		return address;
	}

	public void setAddress(String address) {
		this.address = address;
	}

	public Short getAgent() {
		return agent;
	}

	public void setAgent(Short agent) {
		this.agent = agent;
	}
	
	public String getNumber() {
		return number;
	}

	public void setNumber(String number) {
		this.number = number;
	}

	public static long getSerialversionuid() {
		return serialVersionUID;
	}

}

File: xauth-web/src/main/resources/META-INF/resources/xauth/js/manage/auth_log.js
/**
 * Author:   jhat
 * Date:     2014-01-27
 * Email:    cpf624@126.com
 * Home:     http://pfchen.org
 * Describe: 认证日志管理
 */

Ext.define('XHome.XAuth.Manage.AuthLog', {
    extend: 'XHome.Dashboard.WorkPanel',
    constructor: function(config) {
        if (!config) {
            config = {};
        }

        // 搜索面板
        var spanel = Ext.create('XHome.Dashboard.SearchPanel', {
            items: [{
                name: 'parameters["user"]',
                fieldLabel: '用户',
                labelWidth: 30,
                maxLength: 20,
                maxLengthText: '用户名不能超过20个字符',
                regex: /^[\w-_]+$/,
                regexText: '用户名格式错误',
            }, {
                name: 'parameters["method"]',
                fieldLabel: '认证方式', 
                labelWidth: 55,
                maxLength: 10,
                maxLengthText: '认证方式不能超过10个字符',
                margin: '0 0 0 10',
                disabled: true,
                hidden: true,
            }, {
                name: 'parameters["address"]',
                fieldLabel: '认证地址',
                labelWidth: 55,
                maxLength: 16,
                maxLengthText: '认证地址不能超过16个字符',
                margin: '0 0 0 10',
            }, {
                name: 'parameters["created"]',
                fieldLabel: '认证时间',
                labelWidth: 55,
                maxLength: 20,
                maxLengthText: '认证时间不能超过20个字符',
                margin: '0 0 0 10',
            }],
        });

        // 数据显示表格
        var grid = Ext.create('XHome.Dashboard.EditorGridPanel', {
            autoSelModel: false,
            columns: [{
                text: '编号',
                dataIndex: 'id',
                width: 8,
                hidden: true,
            }, {
                text: '用户',
                dataIndex: 'user',
                width: 20,
                renderer: function(value) {
                    return value.name; 
                },
            }, {
                text: '认证方式',
                dataIndex: 'method',
                width: 25,
                hidden: true,
            }, {
                text: '认证地址',
                dataIndex: 'address',
                width: 20,
            }, {
                text: '认证时间',
                dataIndex: 'createdStr',
                width: 25,
            }, {
                text: '认证结果',
                dataIndex: 'status',
                width: 20,
                renderer: function(value) {
                    return {1: '认证成功', 5: '用户被锁定', 7: '用户不存在',
                        9: '密码错误', 10: '验证码错误', 11: '禁止认证'}[value]; 
                },
            }, {
                text: '设备类型',
                dataIndex: 'agent',
                width: 20,
                renderer: function(value) {
                    return {0: 'OTHER', 1: 'IE', 2: 'CHROME', 3: 'FIREFOX',
                        4: 'OPERA', 5: 'SAFARI', 6: 'BAIDU', 7: 'QIHU',
                        8: 'SOGOU', 9: 'MAXTHON', 10: 'TT', 11: 'UC',
                        12: 'DORPHIN', 50: 'ANDROID', 51: 'IOS',
                        52: 'WINPHONE', 53: 'SYMBIAN'}[value]; 
                },
            }, {
                text: '访问设备',
                dataIndex: 'number',
                renderer: function(value, meta, record) {
                    meta.tdAttr = 'data-qtip="' + value + '"';
                    return value;
                },
            }],
            store: Ext.create('XHome.data.JsonStore', {
                fields: ['id', 'user', 'method', 'address', 'agent', 'number',
                    'createdStr', 'modifiedStr',
                    'owner', 'modifier', 'version', 'status'],
                url: 'xauth/auth_log/query.json',
            }),
        });

        config.items = [spanel, grid];
        this.callParent([config]);
    },
});
